### PR TITLE
Current user canonical

### PR DESCRIPTION
**Pull request by deepakDatametica**
Unparse CURRENT_USER for Hive and BigQuery.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -220,6 +222,14 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
       writer.literal("INT64");
       writer.endFunCall(castFrame);
       break;
+    case SYSTEM_FUNCTION:
+      if (call.getOperator().equals(CURRENT_USER)) {
```
Comment (stutigupta028) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: This seems to be a common code in all 3 dialects. Move this unparse logic to sqlLibraryOperators

File: core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java [hunk 2]
```diff
@@ -165,6 +167,14 @@ public HiveSqlDialect(Context context) {
     case FORMAT:
       unparseFormat(writer, call, leftPrec, rightPrec);
       break;
+    case SYSTEM_FUNCTION:
+      if (call.getOperator().equals(CURRENT_USER)) {
```
Comment (stutigupta028) [inline core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java]: This seems to be a common code for hive and spark. Move this unparse logic to SqlLibraryOperators

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -4233,6 +4233,40 @@ private void checkLiteral2(String expression, String expected) {
     });
   }
 
+  @Test public void testCurrentUser() {
+    String query = "select CURRENT_USER from \"product\" where \"product_id\" = 1";
+    final String expectedSql = "SELECT CURRENT_USER() CURRENT_USER\n"
```
Comment (stutigupta028) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Does CURRENT_USER as alias work on hive and spark without identifier quotes???

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 4]
```diff
@@ -4233,6 +4233,40 @@ private void checkLiteral2(String expression, String expected) {
     });
   }
 
+  @Test public void testCurrentUser() {
+    String query = "select CURRENT_USER from \"product\" where \"product_id\" = 1";
```
Comment (stutigupta028) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: you can simply write  select CURRENT_USER without from clause here

Comment (deepakDatametica) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: done.

File: core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java (hunk 2 above)
Comment (deepakDatametica) [reply core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java]: done.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 3 above)
Comment (deepakDatametica) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: => For hive, current_user is working fine.
=> For spark current_user we are getting below exception:Error: org.apache.spark.sql.AnalysisException: Undefined function: 'current_user'. This function is neither a registered temporary function nor a permanent function registered in the database 'default'.; line 1 pos 7 (state=,code=0)
Hence removed spark current_user changes.

**Pull request merged**